tests/utils/admincli/vmgroupmgmt.go

```diff
@@ -18,27 +18,55 @@
 package admincli
 
 import (
+	"fmt"
 	"log"
 	"strings"
-	"fmt"
+
 	"github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
 )
 
 // CreateVMgroup method is going to create a vmgroup and adds vm to it.
 func CreateVMgroup(ip, name, vmName, dsName string) (string, error) {
 	log.Printf("Creating a vmgroup [%s] on esx [%s]\n", name, ip)
-	return ssh.InvokeCommand(ip, admincli.CreateVMgroup+name+" --default-datastore="+dsName+admincli.VMlist+vmName)
+	out, err := ssh.InvokeCommand(ip, admincli.CreateVMgroup+name+" --default-datastore="+dsName+admincli.VMlist+vmName)
+
+	if err != nil {
+		return out, err
+	}
+
+	// Verify the vmgroup creation
+	if IsVmgroupPresent(ip, name) {
+		return out, nil
+	}
+
+	msg := "Could not verify presence of vmgroup " + name + "  on esx " + ip
+	return msg, fmt.Errorf(msg)
+
 }
 
 // DeleteVMgroup method deletes a vmgroup and removes its volumes as well if "delete_vol" is set
 func DeleteVMgroup(ip, name string, delete_vol bool) (string, error) {
 	log.Printf("Deleting a vmgroup [%s] on esx [%s] with delete_vol[%d]\n", name, ip, delete_vol)
+	var out string
+	var err error
 	if delete_vol {
-		return ssh.InvokeCommand(ip, admincli.RemoveVMgroup+name+admincli.RemoveVolumes)
+		out, err = ssh.InvokeCommand(ip, admincli.RemoveVMgroup+name+admincli.RemoveVolumes)
 	} else {
-		return ssh.InvokeCommand(ip, admincli.RemoveVMgroup+name)
+		out, err = ssh.InvokeCommand(ip, admincli.RemoveVMgroup+name)
+	}
+
+	if err != nil {
+		return out, err
+	}
+
+	// Verify the vmgroup removal
+	if IsVmgroupPresent(ip, name) != true {
```

Comment:
Currently this util doesn't have knowledge about which volumes fall into the given VMgroup. So directly we cannot verify.
What we can do is retrieve the list of volumes from admin cli and in verification make sure those volumes are deleted. But I would prefer doing it out of this PR depending on the need of such verification.